### DISCUSSION TITLE
Continue the discussion: Using fully-qualified function calls

### DISCUSSION THREAD

**Discussion by allowing** [laravel/framework#37590] [Ideas]
Seeing this `PR` is very intriguing. #37517 @claudiodekker

Below is the simplest test.

It is indeed necessary to fully qualify the function. Incomplete qualification requires additional parsing.

Before
![微信截图_20210604140551](https://user-images.githubusercontent.com/20262949/120754000-a281ec80-c53e-11eb-82d5-23606dfcaf74.png)

After
![微信截图_20210604140720](https://user-images.githubusercontent.com/20262949/120754006-a44bb000-c53e-11eb-8077-d330a116064d.png)

**Comment by crynobone** [comment]
With `laravel/pint` you can add the following options to your application:

```json
{
    "preset": "laravel",
    "rules": {
        "native_function_invocation": {
            "include": ["@compiler_optimized"],
            "scope": "namespaced",
            "strict": true
        }
    }
}
```
